Upgrade iOS 16 sleep data

srongch opened the pull request.

Commit:
[Health] Bug fix of #553

srongch committed:
Adapt new sleep data for iOS 16.

- remove the SLEEP_ASLEEP data type as of iOS 16 and unify it with the use SLEEP_ASLEEP_UNSPECIFIED data type
- ignore data types that are not available in the pre-defined data type array

srongch committed:
add sleep analysis data for iOS 16

- change SLEEP_ASLEEP to SLEEP_ASLEEP_UNSPECIFIED
- add helper function for getting all health data types per platform

srongch committed:
Test new changes

- add share function to export data

fix92 commented on packages/health/ios/Classes/SwiftHealthPlugin.swift:
```diff
@@ -175,14 +178,17 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
+                    result(success)
+                    return
+                }
+            }else{
```
```suggestion
            } else {
```

fix92 commented on packages/health/ios/Classes/SwiftHealthPlugin.swift:
```diff
@@ -724,13 +747,20 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
+            dataTypesDict[SLEEP_ASLEEP_CORE] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
+            dataTypesDict[SLEEP_ASLEEP_DEEP] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
+            dataTypesDict[SLEEP_ASLEEP_REM] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
+            //            dataTypesDict[SLEEP_ASLEEP_UNSPECIFIED] = HKSampleType.categoryType(forIdentifier: .sleepAnalysis)!
```
I guess we should remove this comment then

fix92 reviewed (changes requested):
two other things:
1. in the constructor at lib/src/health_data_point.dart:17 there are some special type conversions from seconds to minutes. Do we need this as well for the new types?
2. we should add some documentation in the README.md, so removing SLEEP_ASLEEP from ios and adding three entries for the new ones that are only available on ios and also add the comment that they require the watch

fix92 committed:
Merge pull request #5 from cph-cachet/master

[Health] syncing branch with latest version from cph-cachet/flutter-plugins:master

srongch committed:
Merge branch 'master' into upgrade-ios-health-data

srongch committed:
Update packages/health/ios/Classes/SwiftHealthPlugin.swift

Co-authored-by: Felix <fix92@users.noreply.github.com>

srongch committed:
update health data availability in README

srongch commented:
> two other things:
> 
> 1. in the constructor at lib/src/health_data_point.dart:17 there are some special type conversions from seconds to minutes. Do we need this as well for the new types?
> 2. we should add some documentation in the README.md, so removing SLEEP_ASLEEP from ios and adding three entries for the new ones that are only available on ios and also add the comment that they require the watch

1. Yes, we need that, I think. I updated it. 
2. Updated the README
